lib/language_list.ex

```diff
@@ -0,0 +1,29 @@
+defmodule LanguageList do
+  @moduledoc false
+
+  def new do
+    []
+  end
+
+  def add(list, lang) do
+     [lang | list]
+  end
+
+  def remove(list) do
+    [_ | tail] = list
+    tail
+  end
+
+  def first(list) do
+    [hd | _] = list
+    hd
+  end
+
+  def count(list) do
+    length(list)
+  end
```

Comment:
This is not fair :) [`hd/1`](https://hexdocs.pm/elixir/Kernel.html#hd/1) and [`tl/1`](https://hexdocs.pm/elixir/Kernel.html#tl/1) are also provided by the core.
```suggestion
  def count(list, count \\ 0) # function head, required to supply default args for multiple clauses
  
  def count([], count), do: count
  def count([_|tail], count), do: count(tail, count + 1)
```